snowpack/src/sources/local.ts

```diff
@@ -49,8 +49,13 @@ const NEVER_PEER_PACKAGES: string[] = [
   'es-abstract',
   'node-fetch',
   'whatwg-fetch',
+  'tslib',
 ];
 
+function isPackageEsm(packageManifest: any): boolean {
+  return !!(packageManifest.module || packageManifest.exports || packageManifest.type === 'module');
```

Comment:
Edge case: Looking at [node docs](https://nodejs.org/api/packages.html#packages_exports) I don't think just the existence of an exports field is enough to prove that the package should be treated as esm.

> It is supported in Node.js 12+ as an alternative to the "main" that can support defining subpath exports and conditional exports while encapsulating internal unexported modules.

I think the both of the following should be treated as exposing commonjs:

```json
{"exports": "./main.js", "type": "commonjs"}
```

```json
{"exports": "./main.cjs", "type": "module"}
```

I think a comprehensive check for this to take into account the type field and the extension used in the exports field.

I think this gets further complicated as "is this ESM" is something that needs to happen on a single entrypoint level rather than across a whole package. But we this seems like a larger refactor this can totally be punted. E.g. I could create exports where one entrypoint is exposed as esm, and another entrypoint is cjs only:

```
 "exports": {
    ".": "./main.cjs", // cjs only
    "./subpath": "./secondary.mjs", // mjs only
    "./another-subpath": "./secondary.js", // To know if this is esm/cjs, check the type field
}
```

I could probably take this to pathological obnoxiousness by then introducing conditional exports, but I'm already falling too far down the rabbit hole for this PR :)

Reply:
Good call, even if unlikely you're right that it's not impossible. Luckily I think that we can remove this entirely now with the better CJS handling in place.